decidim-core/lib/decidim/abilities/participatory_process_role_user.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Abilities
+    # Base class used for any participatory process role ability.
+    class ParticipatoryProcessRoleUser
+      include CanCan::Ability
+
+      def initialize(user, context)
+        @user = user
+        @context = context
+
+        # Define abilities if the user is not an admin and has at least one process to manage.
+        if not_admin? && has_manageable_processes?
+          define_abilities
+          define_participatory_process_abilities if current_participatory_process && can_manage_process?(current_participatory_process)
+        end
+      end
+
+      # Grant access to admin panel by default.
+      def define_abilities
+        can :read, :admin_dashboard
```

Comment:
Not every single user because this is only applying for abilities who inherit this class. There is an extra check in the constructor as well:

```ruby
# Define abilities if the user is not an admin and has at least one process to manage.
        if not_admin? && has_manageable_processes?
          define_abilities
          define_participatory_process_abilities if current_participatory_process && can_manage_process?(current_participatory_process)
        end
```